src/app/base/components/bricks-integrator/bricks-integrator.component.ts

```diff
@@ -143,6 +143,10 @@ export class BricksIntegratorComponent implements OnInit, OnDestroy {
     const splitVal: string[] = value.split(',');
     for (let i = 0; i < splitVal.length; i++) {
       filter += "&filters[" + attribute + "][$eq]=" + splitVal[i].trim();
+      // Remove active learners from generators (on ac page we have generators and classifiers but we want to exclude active learners there)
+      if (splitVal[i].trim() == 'generator' && this.executionTypeFilter != "activeLearner") {
+        filter += "&filters[executionType][$ne]=activeLearner";
+      }
```

Comment:
Just an idea but in no means objective better: 

Line 145-148 could be outsourced in an own method with strong name in the signature so I could be straight up understandable what in the for loop happens.

Something e.g.

for [...] {
  filter += createFilterCriteria(attribute)
}